### PR TITLE
Clean up testing 

### PR DESCRIPTION
**Pull request by musm**
supersedes https://github.com/JuliaIO/HDF5.jl/pull/411

### CODE REVIEW

File: test/external.jl [hunk 1]
```diff
@@ -45,4 +48,7 @@ close(source_file)
 # @test read(group2["1"])==1
 # @test read(group2["1.1"])==1.1
 
+rm(fn1)
+# rm(fn2)
```
Comment (musm) [inline test/external.jl]: @ggggggggg trying to rm(fn2) here results is a `unlink: resource busy or locked (EBUSY)` error even though the handle has been explicitly closed, any ideas here?

**Comment by musm** [comment]
will merge sans objections in a few days

File: test/external.jl (hunk 1 above)
Comment (musm) [inline test/external.jl]: probably related to the lib not properly detecting the closed handle

File: test/external.jl [hunk 2]
```diff
@@ -3,46 +3,52 @@ using Base.Test
...
+# these tests succeed
+# reopening the target_file crashes due to "file close degree doesn't match"
+# target_file = h5open(target_file.filename, "r")
+# group2 = target_file["target_group"]["new_group"]
+# @test read(group2["abc"])=="abc"
+# @test read(group2["1"])==1
+# @test read(group2["1.1"])==1.1
+
+rm(fn1)
+# rm(fn2)
```
Comment (tkelman) [inline test/external.jl]: what was preventing this from working? does the file need to manually be closed first?

Comment (musm) [inline test/external.jl]: The problem is that even though it is manually closed, rm claims it is locked....

Comment (tkelman) [inline test/external.jl]: might need to force finalize, or set to nothing and run gc a few times, or explicitly un-mmap more carefully. not a big deal, but good to get resources less leaky

Comment (musm) [inline test/external.jl]: I think this is probably a bug in HDF5.jl, where a resource is not properly freed... Probably will need a simple MWE of the bug, but I haven't found something that exposes in a reduced case.

**Pull request merged**